### PR TITLE
[TwigBundle] use the new Twig autoescaping strategy

### PR DESCRIPTION
**Pull request by fabpot**
| Q | A |
| --- | --- |
| Bug fix? | no |
| New feature? | no |
| BC breaks? | no |
| Deprecations? | no |
| Tests pass? | yes |
| Fixed tickets | n/a |
| License | MIT |
| Doc PR | n/a |

### CODE REVIEW

File: src/Symfony/Bridge/Twig/composer.json [hunk 1]
```diff
@@ -18,7 +18,7 @@
     "require": {
         "php": ">=5.3.9",
         "symfony/security-csrf": "~2.6|~3.0.0",
-        "twig/twig": "~1.13,>=1.13.1"
+        "twig/twig": "~1.17"
```
Comment (fabpot) [inline src/Symfony/Bridge/Twig/composer.json]: We need to release Twig 1.17 first to make the tests pass.

File: src/Symfony/Bundle/TwigBundle/TwigDefaultEscapingStrategy.php [hunk 2]
```diff
@@ -1,37 +0,0 @@
...
- * For the full copyright and license information, please view the LICENSE
- * file that was distributed with this source code.
- */
-
-namespace Symfony\Bundle\TwigBundle;
-
-/**
- * @author Fabien Potencier <fabien@symfony.com>
- */
-class TwigDefaultEscapingStrategy
```
Comment (stof) [inline src/Symfony/Bundle/TwigBundle/TwigDefaultEscapingStrategy.php]: Removing this class is a BC break

Comment (fabpot) [inline src/Symfony/Bundle/TwigBundle/TwigDefaultEscapingStrategy.php]: No, it's not as it was introduced in 2.7.

Comment (fabpot) [inline src/Symfony/Bundle/TwigBundle/TwigDefaultEscapingStrategy.php]: nvm, indeed, it was there before.

Comment (peterrehm) [inline src/Symfony/Bundle/TwigBundle/TwigDefaultEscapingStrategy.php]: And now you are adding deprecations. You might want to add this in the PR header.

Comment (stof) [inline src/Symfony/Bundle/TwigBundle/TwigDefaultEscapingStrategy.php]: it has been added in 2.3.24 when fixing a bug

File: src/Symfony/Bridge/Twig/composer.json (hunk 1 above)
Comment (stof) [inline src/Symfony/Bridge/Twig/composer.json]: I suggest waiting until Twig 1.17 is released before merging this then

Comment (stof) [inline src/Symfony/Bridge/Twig/composer.json]: and the root composer.json also needs to be updated with the new Twig constraint

Comment (fabpot) [inline src/Symfony/Bridge/Twig/composer.json]: fixed

**Pull request merged**